hardware/tests/test_gps_pi.py

```diff
@@ -0,0 +1,148 @@
+from django.test import SimpleTestCase
+from unittest.mock import patch
+from testfixtures import TempDirectory
+
+import os
+
+from hardware.Utils.logger import Logger
+from hardware.gpsPi.gps_reader import GPSReader
+
+
+@patch("serial.Serial")
+class GPSPiTests(SimpleTestCase):
+    def setUp(self):
+        self.temp_dir = TempDirectory()
+
+    def tearDown(self):
+        self.temp_dir.cleanup()
+
+    def test_init_no_logs(self, mock_port):
+
+        # Replace real object os.environ with mock dictionary
+        with patch.dict(
+            os.environ,
+            {
+                "GPS_LOG_FILE": "logger.txt",
+                "LOG_DIRECTORY": self.temp_dir.path,
+                "GPS_PORT": "/dev/serial0",
+                "GPS_BAUDRATE": "9600",
+            },
+        ):
+            gps_reader = GPSReader()
+            mock_port.assert_called_with(
+                os.environ["GPS_PORT"], os.environ["GPS_BAUDRATE"],
+            )
+
+            self.assertTrue(gps_reader.logging is not None)
+            self.assertTrue(gps_reader.logging.name == "GPS_LOG_FILE")
+            self.assertIsInstance(gps_reader.logging, Logger)
+
+    def test_init_logs(self, mock_port):
+
+        with patch.dict(
+            os.environ,
+            {
+                "GPS_HAT_LOG_FILE": "logger.txt",
+                "LOG_DIRECTORY": self.temp_dir.path,
+                "GPS_PORT": "/dev/serial0",
+                "GPS_BAUDRATE": "9600",
+            },
+        ):
+
+            gps_reader = GPSReader("GPS_HAT_LOG_FILE")
+            mock_port.assert_called_with(
+                os.environ["GPS_PORT"], os.environ["GPS_BAUDRATE"],
+            )
+
+            self.assertTrue(gps_reader.logging is not None)
+            self.assertTrue(gps_reader.logging.name == "GPS_HAT_LOG_FILE")
+            self.assertIsInstance(gps_reader.logging, Logger)
+
+    @patch("hardware.gpsPi.gps_reader.date_str_with_current_timezone")
```

Comment:
it doesn't look like the date_str_with_current_timezone function is actually used in gps_reader, is this for the `data["date"]` object at line 52?

Reply:
Yes, it is for the data["date"] object. Trying to understand what is wrong here.

Reply:
i think you're data might be out of date with hardware-team, try running updating your remotes and then run either `git merge origin/hardware-team` or `git rebase origin/hardware-team`

Reply:
okay

Reply:
always a good idea before you're going to PR to update your remotes and check to see if there is new code in your base branch (hardware-team here)

Reply:
Agreed.